zerver/context_processors.py

```diff
@@ -135,6 +135,12 @@ def zulip_default_context(request: HttpRequest) -> Dict[str, Any]:
         settings_path = "/etc/zulip/settings.py"
         settings_comments_path = "/etc/zulip/settings.py"
 
+    self_hosted_signup_page = False
+    if (
+        settings.DEVELOPMENT or not realm or (realm.plan_type == Realm.PLAN_TYPE_SELF_HOSTED)
+    ) and any(keyword in request.get_full_path() for keyword in ["login", "register"]):
```

Comment:
This only conflicts with `calls/zoom/register` endpoint, which doesn't seem to render footer anyway.

Reply:
This is really fragile. You can pass additional variables into the context for specific pages by including  `zproject/urls.py`, for example:
```
    path(                                                                                                
        "accounts/new/send_confirm/<email>",                                                             
        TemplateView.as_view(template_name="zerver/accounts_send_confirm.html"),                         
        {"realm_creation": True},                                                                        
        name="new_realm_send_confirm",                                                                   
    ),                                                                                                   
```

and maybe one could have the template do an OR of whether it's a signup page with whether you're self-hosted (computed here). But I think we actually don't care whether it's a signup page -- we don't want the zulip.com footer on the password reset form (or any other portico pages) for self-hosted servers, so I think the variable we want to add here is just `self_hosted` and have that control the footer situation.

Unfortunately, I think we can't cleanly do that without first removing the zulip.com pages from self-hosted servers, which is something we've wanted to do for some time, so maybe we should just apply some focus to doing that. This is https://github.com/zulip/zulip/issues/15396 -- do you want to pick it up? I'll post a long comment on how to process shortly.

Reply:
Actually, scratch that -- I'll post a comment at the bottom of this issue, but I think I had the direction wrong: this PR needs to happen before #15396, not the other way around. Thinking about this more, I think we just need to make the shape of the `footer.html` determined solely by the `corporate_enabled=settings.CORPORATE_ENABLED` setting, which we currently pass directly into a couple templates, but could readily add to the common context. So the proposal is:

(1) Replace this bit of logic with just adding `corporate_enabled` to `zulip_default_context`, and having `footer.html` check that.
(2) Figure out an alternative CSS solution for managing page height that's going to work on all of our non-landing portico pages (the password reset flow, /accounts/find, /accounts/next, etc.). I imagine the `self-hosted-register-page` class of approach is not really going to work for this goal.

Once we've done that, we can then freely bulk-move the landing pages from the `zerver` app to the `corporate` app, since with the footer removed, we won't generally have any incoming links to landing pages from `zerver` pages.